subsys/bluetooth/services/fast_pair/fp_storage/include/fp_storage_ak_bond.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _FP_STORAGE_AK_BOND_H_
+#define _FP_STORAGE_AK_BOND_H_
+
+#include <sys/types.h>
+#include <stdbool.h>
+#include <zephyr/bluetooth/conn.h>
+
+#include "fp_common.h"
+
+/**
+ * @defgroup fp_storage_ak_bond Fast Pair storage of associations between Account Keys and bonds
+ * @brief Internal API for Fast Pair storage of associations between Account Keys and bonds
+ *
+ * Used only when the CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND Kconfig option is enabled.
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Create and save Account Key association with the connected potential Bluetooth bond.
+ *
+ *  Used only when the CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND Kconfig option is enabled.
+ *
+ *  The module links the Account Key information with the Bluetooth bond and stores
+ *  it in the NVM memory. The association and the Bluetooth bond will be deleted if the bond is not
+ *  confirmed using the @ref fp_storage_ak_bond_conn_confirm API before the bonding process is
+ *  cancelled using the @ref fp_storage_ak_bond_conn_cancel API.
```

Comment:
That's not true. We call `fp_storage_ak_bond_conn_cancel` on disconnection. Then we check if the bond is confirmed. If yes - do nothing. Otherwise delete it.

Reply:
the word "cancel" here is a bit controversial but we decided it's better than "stopped", because it clearly indicates that it won't be resumed.

Reply:
`conn_finalize` as this API is used in the success case and the failure case. CC: @MarekPieta

Reply:
`finalize` may not be the most fortunate word too (it's actually not needed to finish the store operation). Maybe we should simply avoid calling it once we confirmed bond in `fp_keys`? That may make code more intuitive.

Reply:
I'd let the storage module decide what to do with the bond after the conn is disconnected. Let's call it `finalize` if nothing betters comes to mind.